### PR TITLE
cli: add helm customization flags to core install

### LINKED ISSUE #5506: cli: Add helm customization flags to linkerd install cmd
## Feature Request

Allow the helm way of customization through `set`, `set-string`, `values`, `set-files` flags for `linkerd install` cmd. this should follow the same extension install model

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #5506

This branch adds helm way of customization through
 `set`, `set-string`, `values`, `set-files` flags for
the core control-plane install and upgrade cmds.

For this to work, the helm v3 engine rendering helpers
had to be used instead of our own wrapper type.

## Testing

```bash
tarun in dev in linkerd2 on  tarun/install-helm-flagss [$!?] via 🐹 v1.15.4 took 4s
❯ ./bin/go-run cli install --ignore-cluster --set global.namespace=l6d | grep l6d
WARN[0000] could not set namespace from kubectl context, using 'default' namespace: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
                 ensure the KUBECONFIG path  is valid
  name: l6d
    linkerd.io/control-plane-ns: l6d
  name: linkerd-l6d-identity
    linkerd.io/control-plane-ns: l6d
  name: linkerd-l6d-identity
    linkerd.io/control-plane-ns: l6d
  name: linkerd-l6d-identity
  namespace: l6d
.
.
.
```

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: cli/cmd/install.go [hunk 1]
```diff
@@ -320,20 +332,61 @@ func render(w io.Writer, values *l5dcharts.Values, stage string) error {
...
+		return err
+	}
+
+	// Create values override
+	valuesOverrides, err := options.MergeValues(nil)
+	if err != nil {
+		return err
+	}
+
+	vals, err := chartutil.CoalesceValues(chart, valuesOverrides)
```
Comment (adleong) [inline cli/cmd/install.go:370]: If I'm reading this correctly, this matches what I describe in https://github.com/linkerd/linkerd2/issues/5158#issuecomment-758306503.  i.e. `MergeValues` will return a set of value overrides generated from the `--set` and `--values` flags and then these overrides will be applied on top of the values loaded from the the Linkerd cluster.

File: cli/cmd/testdata/install_config.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+global:
+  clusterDomain: example.com
+  namespace: l5d
+
+installNamespace: false
```
Comment (adleong) [inline cli/cmd/testdata/install_config.yaml:5]: newline please

File: cli/cmd/install.go (hunk 1 above)
Comment (Pothulapati) [reply cli/cmd/install.go:370]: Yes, That is right as `chart.Values` contains the (default+current cluster config+CLI custom flags) while `vals` is through helm customization flags.

**Review by adleong — COMMENTED** [review]
I think the `--config` flag, defined in `options.go` should be removed in favor of the `--values` flag.

**Review by adleong — APPROVED** [review]

File: cli/cmd/install.go [hunk 3]
```diff
@@ -320,20 +336,61 @@ func render(w io.Writer, values *l5dcharts.Values, stage string) error {
...
-		Files:     files,
-		Fs:        static.Templates,
+	var partialFiles []*loader.BufferedFile
+	for _, template := range charts.L5dPartials {
+		partialFiles = append(partialFiles,
+			&loader.BufferedFile{Name: template},
+		)
+	}
+
+	// Load all Viz chart files into buffer
```
Comment (alpeb) [inline cli/cmd/install.go:346]:
```suggestion
	// Load all chart files into buffer
```

**Review by alpeb — COMMENTED** [review]
Ability to bring Helm configurability to the CLI: huge win!! :partying_face:

**Comment by Pothulapati** [comment]
@adleong Should we have these flags for `linkerd install config`? I'm wondering if it could be confusing to users as most values in the readme would not be useful in that case.

**Comment by adleong** [comment]
All stages of `linkerd install` use the same `values.yml`, they just render out different templates.  So yes, I think the flags should be available to `linkerd install`, `linkerd install config`, and `linkerd install control-plane`.

**Comment by Pothulapati** [comment]
Updated!

**Review by kleimkuhler — APPROVED** [review]
👍

**Pull request merged**